### ISSUE TITLE
[Bug]: Batch inference Olive exported models

### ISSUE THREAD

**Issue by Rishit-dagli** [microsoft/Olive#354]
### What happened?

I was trying to get a Whisper model exported with ONNX run with batch inference on which I notice that the exported model is equipped to only handle `(1, x)` shaped inputs and so I modify the inputs by running:

```py
model.graph.input[0].type.tensor_type.shape.dim[0].dim_param = 'batch_size'
model.graph.input[0].type.tensor_type.shape.dim[1].dim_param = 'sequence_length'
model.graph.input[0].type.tensor_type.shape.dim[1].ClearField('dim_value')
model.graph.input[0].type.tensor_type.shape.dim[0].ClearField('dim_value')   
```

and then inference with the newly saved model which gives me:

> onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running AudioDecoder node. Name:'AudioDecoder_1' Status Message: [AudioDecoder]: Expect input dimension [n] or [1,n].

This gives me the impression that the Audio Decoder is not equipped to handle batch inputs. I also tried to work with a model with no audio decoder and do that using `librosa` which does not seem to work well too.

I was wondering if I could get any help or anything I might have missed in exporting the model?

### Version?

`0.2.1`, I am yet to test with the latest commits

**Comment by xiaoyu-work** [comment]
Hi @Rishit-dagli , what error you got without using audio decoder? Also in the latest commit, we added detailed codes about custom components for Whisper model. Feel free to update them to meet your requires.

**Comment by Rishit-dagli** [comment]
@xiaoyu-work Ah, yes, missed that:

This is what I see without using audio decoder:

> onnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Non-zero status code returned while running BeamSearch node. Name:'BeamSearch_node' Status Message: Input 'attention_mask' is expected to have same shape as input_ids

> Also in the latest commit, we added detailed codes about custom components for Whisper model. Feel free to update them to meet your requires.

The model exported with the latest commit still expects `(1, x)` shaped inputs as I see from this message, do you think I might have missed something? Let me also try modifying the inputs manually as I did earlier with this version of the model.

> onnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Got invalid dimensions for input: audio_stream for the following indices

**Comment by wenbingl** [comment]
@Rishit-dagli , Right now, the audio decoder in onnxruntime-extensions only supports 1-D audio input. Can you share more details about how you create the N-dim batch audio inputs?

**Comment by Rishit-dagli** [comment]
> Right now, the audio decoder in onnxruntime-extensions only supports 1-D audio input

Ah, I see!

I also tried without the audio decoder so I load the data using:

```py
audio_blob, _ = librosa.load(path)
audio_blob = np.expand_dims(audio_blob, axis=0)
```

This gets me `(1,x)` sized array which I then rearrange to form `(x//b, b)` sized array and do the padding. Running the model now without the audio decoder seems to be able to do inference but gives outputs of the pattern:

> first batch transcribed well
> transcribed well !!!!!!!!!!!!!!!
> !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

After the first batch it starts returning exclamation marks and I have not been able to understand what causes this.

**Comment by wenbingl** [comment]
In that case,  I don't think the batch input can solve your problem, check openai examples (https://github.com/openai/whisper/blob/main/whisper/transcribe.py) to see how to process the long file transcribing.

**Comment by jambayk** [comment]
Closing this issue since it is not an Olive issue and has also turned stale.
